tee-worker/litentry/core/vc-task/receiver/src/lib.rs

```diff
@@ -195,8 +196,22 @@ pub fn run_vc_handler_runner<ShieldingKeyRepository, A, S, H, O, Z, N>(
 				);
 
 				// Totally fine to `unwrap` here. Because new item was just added above.
-				let do_watch = req_registry_pool.update_item(connection_hash).unwrap();
-				send_vc_response(connection_hash, context_pool, response, 0u8, 1, do_watch);
+				match req_registry_pool.update_item(connection_hash) {
+					Ok(do_watch) => {
+						send_vc_response(connection_hash, context_pool, response, 0u8, 1, do_watch);
+					},
+					Err(e) => {
+						error!("1 couldn't find connection_hash: {:?}", e);
+						send_vc_response(
+							connection_hash,
+							context_pool,
+							Err(format!("1 couldn't find connection_hash: {:?}", e)),
+							0u8,
+							1,
+							false,
+						);
+					},
```

Comment:
It seems no matter `Ok` or `Err` we are going to call `send_vc_response` anyway, only the response and do_watch are different.

Can we have code like this
```
let (response, do_watch) = match req_registry_pool.update_item(connection_hash) { .. };
send_vc_response(....)
```

I think it has a better readability

Reply:
Same to other places

Reply:
`response` is not returned from `update_item`.